cartography/intel/aws/cloudtrail_management_events.py

```
@@ -0,0 +1,472 @@
+    3. Return aggregated relationships ready for loading
+
+    :type events: List[Dict[str, Any]]
+    :param events: List of raw CloudTrail events from lookup_events API
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: List[Dict[str, Any]]
+    :return: List of aggregated role assumption relationships ready for loading
+    """
+    role_assumptions = []
+
+    logger.info(
+        f"Transforming {len(events)} CloudTrail events to role assumptions for region '{region}'"
+    )
+
+    for event in events:
+        # Extract role assumption details from CloudTrail event
+        event_name = event.get("EventName")
+        if not event_name:
+            logger.debug(
+                f"Event missing EventName field: {event.get('EventId', 'unknown')}"
+            )
+            continue
+
+        # Parse CloudTrail event JSON once
+        cloudtrail_event = None
+        if event.get("CloudTrailEvent"):
+            try:
+                cloudtrail_event = json.loads(event["CloudTrailEvent"])
+            except (json.JSONDecodeError, TypeError):
+                logger.debug(
+                    f"Failed to parse CloudTrail JSON for event {event.get('EventId', 'unknown')}"
+                )
+
+        # Simple source principal extraction
+        user_identity = event.get("UserIdentity", {})
+        source_principal = (
+            user_identity.get("arn")
+            or user_identity.get("principalId")
+            or event.get("UserName")
+            or (
+                cloudtrail_event and cloudtrail_event.get("userIdentity", {}).get("arn")
+            )
+        )
+
+        if not source_principal:
+            logger.debug(
+                f"Could not extract source principal from event {event.get('EventId', 'unknown')}"
+            )
+            continue
```

is it really this hard and complex to get a source principal out? Pick one place to get it from and that's it. there is no need to use this many fallbacks at such an early stage of the code.

From the [spec](https://docs.google.com/document/d/1GvOeLB6zC1l0-PPn7wG-etybrP0bwlheKarxkJ_5_bs/edit?tab=t.0#heading=h.bg8k06ncfpe5), I think that's in `requestParameters['roleArn']`.

If it's none or whatever, so be it, let it fail, we'll fix it in a follow up

---

I was able to simplify source_principal to just use simple dict access. 

Im having a little more trouble with getting dest principal to have similar super simple access. Right now what im doing is iterating over the events resources and if i find a ResourceType that equals AWS:IAM:Role then I use dict access to assign that to destination_principal, but this still seems like it is too complex. 
  
Also when i run cartography an error pops up that says that some events are missing fields so I put a try except in this section so that it skips those events and continues the sync.

It would be helpful if we could look at this section of code together to see where I'm going wrong.